build.rs

```diff
@@ -0,0 +1,27 @@
+use std::process::Command;
+use std::str;
+
+fn main() {
+    // Temporary check to see if the rustc version >= 1.81 in which case the
+    // `Error` trait is always available. This is temporary because in the
+    // future the MSRV of this crate will be beyond 1.81 in which case this
+    // build script can be deleted.
+    let minor = rustc_minor_version().unwrap_or(0);
+    if minor >= 81 {
+        println!("cargo:rustc-cfg=core_error");
+    }
+    if minor >= 80 {
+        println!("cargo:rustc-check-cfg=cfg(core_error)");
+    }
+}
+
+fn rustc_minor_version() -> Option<u32> {
+    let rustc = std::env::var("RUSTC").unwrap();
```

Comment:
This needs to handle RUSTC_WRAPPER too I think.

Reply:
FWIW just looking at `RUSTC` is what [anyhow does](https://github.com/dtolnay/anyhow/blob/8852dc321822fe26b5cc84b2333904c897aff70a/build.rs#L196) which is what this is copied from

Reply:
I guess it is less important for a version check. For a compile you definitively have to respect it and RUSTFLAGS though, which anyhow does.

Reply:
My understanding is that any wrapper must call the rustc that is passed to it, which will be RUSTC, and we are not interested in the version of the wrapper itself, so we would need to instruct the wrapper to call rustc to get its version which is a lot of complication for no gain.